src/wp-includes/block-supports/layout.php

```diff
@@ -883,17 +883,63 @@ function wp_restore_group_inner_container( $block_content, $block ) {
 		return $block_content;
 	}
 
-	$replace_regex   = sprintf(
+	/**
+	 * This filter runs after the layout classnames have been added to the block, so they
+	 * have to be removed from the outer wrapper and then added to the inner.
+	*/
+	$layout_classes = array();
+	$processor      = new WP_HTML_Tag_Processor( $block_content );
+
+	if ( $processor->next_tag( array( 'class_name' => 'wp-block-group' ) ) ) {
+		if ( method_exists( $processor, 'class_list' ) ) {
+			foreach ( $processor->class_list() as $class_name ) {
+				if ( str_contains( $class_name, 'layout' ) ) {
+					array_push( $layout_classes, $class_name );
+					$processor->remove_class( $class_name );
+				}
+			}
+		} else {
```

Comment:
This if-else should be removed since in WordPress core we know for sure the `class_list()` method is present as the code here only affects the latest version anyway. So we should only keep the inner logic from the `if` clause in here.

In Gutenberg, this is only needed because it needs to work with different WordPress versions.

Reply:
Totally missed this when copying the code across 😅